Adds a link to the OOBE What's New Section in the ShellPage Navigation

iakrayna opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request
Adds an "About this App" in the ShellPage that redirects to the What's New section of the OOBE Page.
Links Issue 17611

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [ ] **Closes:** #17611 
- [ ] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [ ] **Tests:** Added/updated and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed

Commit:
Adds a link to the OOBE What's New Section in the ShellPage Navigation

iakrayna commented:
@microsoft-github-policy-service agree company="Microsoft"

crutkas commented on src/settings-ui/Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -2146,6 +2146,9 @@ From there, simply click on one of the supported files in the File Explorer and
     <comment>Don't loc "PowerToys"</comment>
+  </data>
+	<data name="AboutApp_NavViewItem.Content" xml:space="preserve">
+    <value>About This App</value>
```
Should be "What's new"

iakrayna replied:
Fixed per recommendation

crutkas commented on src/settings-ui/Settings.UI/SettingsXAML/Views/ShellPage.xaml:
```diff
@@ -204,6 +204,10 @@
+                    <NavigationViewItem
+                        x:Uid="AboutApp_NavViewItem"
+                        Icon="{ui:FontIcon Glyph=&#xF133;}"
+                        Tapped="AboutItem_Tapped" />
```
"WhatIsNewItem_Tapped"

iakrayna replied:
changed per recommendation

crutkas commented on src/settings-ui/Settings.UI/SettingsXAML/Views/ShellPage.xaml.cs:
```diff
@@ -316,6 +335,11 @@ private async void FeedbackItem_Tapped(object sender, Microsoft.UI.Xaml.Input.Ta
             await Launcher.LaunchUriAsync(new Uri("https://aka.ms/powerToysGiveFeedback"));
         }
 
+        private void AboutItem_Tapped(object sender, Microsoft.UI.Xaml.Input.TappedRoutedEventArgs e)
```
WhatIsNewItem_Tapped

iakrayna replied:
Fixed per recommendation

crutkas commented:
@jaimecbernardo anything else needed?

jaimecbernardo reviewed (changes requested):
Thanks for opening the PR!

Besides the feedback from @crutkas  , I've tested that the button does nothing if the OOBE window is already open, The idea is that it would switch to the "What's new tab", right?

crutkas commented:
Yes it should flip to what’s new IMO. Good catch.

crutkas commented:
@iakrayna is this something you feel you can fix?

iakrayna commented:
@crutkas  I'll have to look into it.  Sorry, it's been a busy time and I haven't cracked open the code.

iakrayna commented:
@crutkas 

Sorry, I know it's been a while.  I managed to fix the verbiage issues - but it looks like there is an issue with App.GetOobeWindow() as it always returns null regardless if the Ooobe Window is open.

That being said, I haven't merged with dev.  So if this was a known issue that's been fixed, I don't know yet.

Commit:
Updated verbiage to code as per recommendation

iakrayna committed:
Merge branch 'microsoft:main' into main

Commit:
Updated so that if the Oobe Window is active, clicking on the What's New option in the ShellPage navigates to the What's New page.

Added a "SetActiveWindow" Method the OobeWindow to easily open an active window of a set PowerToysModule type - this is to support being able to open the What's New Page without having to close the Oobe Window and close it every time.

Commit:
Merge branch 'main' of https://github.com/iakrayna/PowerToys

Commit:
Renamed SetActiveWindow to SetAppWindow

iakrayna commented:
@crutkas

Okay, I synched up with Dev and I was able to create a way to go to the What's New page regardless if the Oobe Window is open.  There was no direct way to open a specific page, so I created a new public method within the OobeWindow class to navigate to any page as long as you pass a PowerToys Module into it.

The other way I was able to get it to work was to actually Close the latest instance of the Oobe Window and Open a new instance that navigated to the What's New page - that didn't seem quite as clean.

Commit:
Adjusted method and property names for naming consistency across the application

jerone commented on src/settings-ui/Settings.UI/SettingsXAML/OobeWindow.xaml.cs:
```diff
@@ -105,6 +105,14 @@ public OobeWindow(PowerToysModules initialModule, bool isDark)
+        {
+            if (shellPage != null)
+            {
+                shellPage.NavigateToModule(this.initialModule);
```
Shouldn't this be `module`, or this method parameter is redundant.

```suggestion
                shellPage.NavigateToModule(module);
```

iakrayna replied:
Good catch - that's what I get for copy/paste mindlessly. 

Fixed.

Commit:
Fixed an oversight.

jaimecbernardo committed:
Merge branch 'main' into pr29028

jaimecbernardo commented:
I've pushed a commit to put this PR up to date with main so that it's buildable on Visual Studio 17.8

jaimecbernardo reviewed (approved):
LGTM! Thank you for the contribution!
